src/main/java/com/powsybl/openloadflow/sa/AbstractSecurityAnalysis.java

```diff
@@ -244,14 +238,119 @@ SecurityAnalysisReport runSync(SecurityAnalysisParameters securityAnalysisParame
+
+                // Merge into first result
+                // PreContingency results first
+                preContingenctViolations.addAll(resultOtherComponent.getPreContingencyResult().getLimitViolationsResult().getLimitViolations());
+                mergedPreContingencyNetworkResult =
+                        mergeNetworkResult(resultOtherComponent.getPreContingencyResult().getNetworkResult(), mergedPreContingencyNetworkResult);
+
+                // PostContingency and OperatorStrategies results
+                mergeSecurityAnalysisResult(resultOtherComponent, postContingencyResults, operatorStrategyResults);
+            }
+        }
+        PreContingencyResult mergedPrecontingencyResult =
+                new PreContingencyResult(result.getPreContingencyResult().getStatus(),
+                        new LimitViolationsResult(preContingenctViolations),
+                        mergedPreContingencyNetworkResult);
+        return new SecurityAnalysisResult(mergedPrecontingencyResult, postContingencyResults, operatorStrategyResults);
+    }
+
+    void mergeSecurityAnalysisResult(SecurityAnalysisResult resultToMerge, List<PostContingencyResult> postContingencyResults,
+                                     List<OperatorStrategyResult> operatorStrategyResults) {
+        resultToMerge.getPostContingencyResults().forEach(postContingencyResult -> {
+            Optional<PostContingencyResult> optOriginalResult = postContingencyResults.stream()
+                    .filter(originalPostContingencyResult -> originalPostContingencyResult.getContingency().getId().equals(postContingencyResult.getContingency().getId()))
+                    .findAny();
+
+            if (optOriginalResult.isPresent()) {
+                PostContingencyResult originalResult = optOriginalResult.get();
+                NetworkResult mergedNetworkResult = mergeNetworkResult(postContingencyResult.getNetworkResult(), originalResult.getNetworkResult());
+                List<LimitViolation> violations = new ArrayList<>(postContingencyResult.getLimitViolationsResult().getLimitViolations());
+                violations.addAll(originalResult.getLimitViolationsResult().getLimitViolations());
+
+                PostContingencyResult mergedPostContingencyResult =
+                        new PostContingencyResult(originalResult.getContingency(), originalResult.getStatus(),
+                                new LimitViolationsResult(violations), mergedNetworkResult, originalResult.getConnectivityResult());
+
+                postContingencyResults.remove(originalResult);
+                postContingencyResults.add(mergedPostContingencyResult);
+            } else {
+                postContingencyResults.add(postContingencyResult);
+            }
+        });
+
+        resultToMerge.getOperatorStrategyResults().forEach(operatorStrategyResult -> {
+            Optional<OperatorStrategyResult> optOriginalResult = operatorStrategyResults.stream()
```

Comment:
This is very slow.
If you have N contingencies, you iterate on average N times (assuming duplicates are rare)
And the loop is performed for each contingency. Total complexity is quadratic vs number of contingencies [ N*(N-1)/2]

It would be better to use a Map.

Reply:
I think the best way to fix that is to make the SecurityAnalysisResult API evolve so it allows us to either efficiently merge results, or store them per component and have an efficient API to query them.
I'll see if I can do better here but I wouldn't over optimize it too much on this side as this is a temporary fix and this logic should not be the permanent fix.

Reply:
Oups. I only saw your statement on not overoptimizing and didn(t see that you said you would try something.

I committed something to reduce iterations and list copies. Let me think if it works for you. Sorry if it conficlted with work you already started.
https://github.com/powsybl/powsybl-open-loadflow/pull/1089/commits/207787f7c2c64a9550f2eba00948115d75569dba

Reply:
No problem, looks good to me.